photon-client/src/components/settings/DeviceControlCard.vue

```diff
@@ -127,9 +127,35 @@ const openExportLogsPrompt = () => {
   exportLogFile.value.click();
 };
 
-const exportSettings = ref();
+const exportInProgress = ref<boolean>(false);
 const openExportSettingsPrompt = () => {
-  exportSettings.value.click();
+  exportInProgress.value = true;
+  axios
+    .get(`http://${address}/api/settings/photonvision_config.zip`, {
+      responseType: "blob"
+    })
+    .then((resp) => {
+      const element = document.createElement("a");
+      element.style.display = "none";
+
+      // var blob = new Blob([resp.data], {type: "application/zip"});
+      const blob = resp.data;
+      element.href = window.URL.createObjectURL(blob);
+
+      element.setAttribute("download", "photonvision_config.zip");
+
+      document.body.appendChild(element);
+      element.click();
+      document.body.removeChild(element);
+    })
+    .catch(() => {
+      useStateStore().showSnackbarMessage({
+        color: "error",
+        message:
+          "Calibration data isn't available for the requested resolution, please calibrate the requested resolution first"
+      });
+    })
+    .finally(() => (exportInProgress.value = false));
```

Comment:
I don't get the point of these changes, if the backend is just going to return a zip file then the <a> tag can be set to download it as is. Also, this doesn't check for 500s codes or if the backend is connected like other axios requests do.

Reply:
yeah fair i wanted a modal to show up while waiting for download to happen but if you see a progress bar anyways i dont see the point either. the cal json download could also be changed to download directly from the backend too? is the modal helpful?

Reply:
yeah revertying